homeassistant/components/lcn/services.py

```diff
@@ -56,18 +55,20 @@ class LcnServiceCall:
     def __init__(self, hass):
         """Initialize service call."""
         self.hass = hass
-        self.connections = hass.data[DATA_LCN][CONF_CONNECTIONS]
 
     def get_device_connection(self, service):
-        """Get device connection object."""
-        addr, connection_id = service.data[CONF_ADDRESS]
-        addr = pypck.lcn_addr.LcnAddr(*addr)
-        if connection_id is None:
-            connection = self.connections[0]
-        else:
-            connection = get_connection(self.connections, connection_id)
+        """Get address connection object."""
+        address, host_name = service.data[CONF_ADDRESS]
 
-        return connection.get_address_conn(addr)
+        for config_entry in self.hass.config_entries.async_entries(DOMAIN):
+            if config_entry.title == host_name:
```

Comment:
I think Martin's comments about config_entry.title apply to this use and others I was asking about.  I'd again suggest adding it to the config entry explicitly rather than using `title` which is for display purposes.

Reply:
I know that it should not be like this and if you still insist, I will add the initial host_name to the config_entry.data.
But I'm so hesitant because the host_name wont be used in the future and we will have unused data in the config_entry.data. Furthermore, if the user changes the entry.title he has no relation to the initial chosen host_name.

I had a look at the [device actions](https://developers.home-assistant.io/docs/device_automation_action/) Martin mentioned. That's really the right thing to go with in this case as there will be no need for any host_name or config_entry.title. But for this I first need the device registry set up properly (one of the next PRs). That will be a breaking change for the user..

Although it's not nice, I suggest we get by with this intermediate solution.

Reply:
Sounds like we got confirmation from Martin that config_entry.title shouldn't be used internally within an integration like this, and is just a user facing text string, and the other constraints should probably be solved some other way.

Reply:
If we want to avoid breaking changes while working on multiple PRs we can use a different target branch than dev branch to make sure we don't release anything until we're ready. Multiple PRs can then be merged in the separate target branch before we decide to release the whole feature range.

Reply:
I added the initital host name to config_entry.data and use this for addressing purposes in the service calls.
I don't think, we need a different target branch.
As soon as there is no longer access to the hostname in config_entry.data in the future, it will silently disappear in the user configurations. :smirk:  I'm fine with that.